### PR TITLE
Fix invalid discount calculaion for specific product voucher that is applied once per order

### PR DESCRIPTION
**Pull request by IKarbowiak**
For a specific product voucher with the "apply once per order" flag set to true, the discount is applied to the cheapest line item, rather than the cheapest line item that the voucher covers.

Fixes https://github.com/saleor/saleor/issues/12107

# Impact

* [ ] New migrations
* [ ] New/Updated API fields or mutations
* [ ] Deprecated API fields or mutations
* [ ] Removed API types, fields, or mutations
* [ ] Documentation needs to be updated

# Pull Request Checklist

* [ ] Privileged queries and mutations are guarded by proper permission checks
* [ ] Database queries are optimized and the number of queries is constant
* [ ] Database migration files are up to date
* [ ] The changes are tested
* [ ] GraphQL schema and type definitions are up to date
* [ ] Changes are mentioned in the changelog

### CODE REVIEW

File: saleor/graphql/checkout/tests/mutations/test_checkout_add_promo_code.py [hunk 1]
```diff
@@ -543,6 +544,78 @@ def test_checkout_add_category_code_checkout_with_sale(
...
+        address=checkout.shipping_address,
+        discounts=[],
+    )
+
+    variables = {
+        "id": to_global_id_or_none(checkout),
+        "promoCode": voucher.code,
+    }
+
+    # wheng
```
Comment (fowczarek) [inline saleor/graphql/checkout/tests/mutations/test_checkout_add_promo_code.py:606]:
```suggestion
    # when
```

File: saleor/graphql/checkout/tests/mutations/test_checkout_add_promo_code.py [hunk 2]
```diff
@@ -543,6 +544,78 @@ def test_checkout_add_category_code_checkout_with_sale(
     assert checkout.discount_amount == Decimal(0)
 
 
+def test_checkout_add_variant_voucher_code_apply_once_per_order(
+    api_client, checkout_with_items, voucher_specific_product_type
+):
+    # TODO: update third variant price - include the third variant in discount
```
Comment (fowczarek) [inline saleor/graphql/checkout/tests/mutations/test_checkout_add_promo_code.py:550]: How about it?

**Review by fowczarek — APPROVED** [review]

File: saleor/graphql/checkout/tests/mutations/test_checkout_add_promo_code.py (hunk 2 above)
Comment (IKarbowiak) [reply saleor/graphql/checkout/tests/mutations/test_checkout_add_promo_code.py:550]: Done, I will remove the comment. Thanks!

File: saleor/checkout/fetch.py [hunk 3]
```diff
@@ -360,33 +360,46 @@ def apply_voucher_to_checkout_line(
...
+            collections=line_info.collections,
+            channel=channel,
+            channel_listing=line_info.channel_listing,
+            discounts=discounts,
+            price_override=line_info.line.price_override,
+        )
+        if not cheapest_line or cheapest_line_price > line_price:
+            cheapest_line_price = line_price
+            cheapest_line = line_info
+    return cheapest_line
```
Comment (kadewu) [inline saleor/checkout/fetch.py:400]: You can use `min` here if you will like it.
```suggestion
    def variant_price(line_info):
        return line_info.variant.get_price(
            product=line_info.product,
            collections=line_info.collections,
            channel=channel,
            channel_listing=line_info.channel_listing,
            discounts=discounts,
            price_override=line_info.line.price_override,
        )

    return min(lines_info, key=variant_price)
```

**Review by kadewu — APPROVED** [review]

**Review by szczecha — APPROVED** [review]

**Pull request merged**